am-lib/src/main/java/uk/gov/hmcts/reform/amlib/enums/Permission.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.amlib.enums;
+
+import uk.gov.hmcts.reform.amlib.exceptions.UnsupportedPermissionsException;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Exposes a set of enum values used to set permissions for Access Management.
+ * Each of the values is a power of two. The reason for that is that in Access Management
+ * there might be multiple permissions that need to be assigned: ie. READ + CREATE + HIDE.
+ * For convenience when saving a record into AM all the values are summed up (the 'sumOf' method) and saved as integer.
+ * In order to determine which individual permissions a record has
+ * the binary 'AND' operation is done (the 'hasPermissionTo' method).
+ */
+public enum Permission {
+    HIDE(0),
+    CREATE(1),
+    READ(2),
+    UPDATE(4),
+    SHARE(8),
+    DELETE(16);
+
+    private int value;
+
+    Permission(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public static int sumOf(Set<Permission> perms) {
```

Comment:
Can we extract this and `buildPermissions` to `Permissions` utility class.

It makes more sense to have:

```
Permissions.sumOf(CREATE, READ)

Permissions.fromSum(3) # or similar
```

Reply:
👍